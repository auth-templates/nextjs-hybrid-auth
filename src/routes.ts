export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}