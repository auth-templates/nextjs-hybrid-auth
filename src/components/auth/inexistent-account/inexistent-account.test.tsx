import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import InexistentAccount from './inexistent-account';
import {PublicRoutes} from '../../../routes';

describe("InexistentAccount", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <InexistentAccount email={'account@mail.com'} />
            </MemoryRouter>
        );

        expect(screen.getByText(/No account with the email/)).toBeInTheDocument(); 
        expect(screen.getByText(/account@mail.com/)).toBeInTheDocument();
        expect(screen.getByText(/has been found. Please go to/)).toBeInTheDocument();
        expect(screen.getByRole('link', {name: 'Sign Up'})).toHaveAttribute('href', PublicRoutes.register)
        expect(screen.getByText(/page in order to create an account./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <InexistentAccount email={'account@mail.com'} /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});