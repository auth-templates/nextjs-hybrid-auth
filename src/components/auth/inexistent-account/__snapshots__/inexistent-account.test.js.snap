// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InexistentAccount renders correctly 1`] = `
<div
  className="inexistentAccount"
>
  No account with the email 
  <div
    className="email"
  >
    account@mail.com
  </div>
   has been found. Please go to
  <a
    href="/register"
    onClick={[Function]}
  >
     Sign Up 
  </a>
  page in order to create an account or try again with a different email.
</div>
`;
